### ISSUE TITLE
aeroscapes

### ISSUE THREAD

**Issue by JAYCHUS** [leoda1/UTFormer#1]
请问训练aeroscapes数据集 要怎么摆放呢

**Comment by leoda1** [comment]
更新了main分支的代码，aeroscapes本身就是VOC格式的数据集，所以你直接按照dataset目录下的放就ok了

**Comment by JAYCHUS** [comment]
好嘞好嘞 谢谢您 还有就是训练的时候除了 修改类别数还需要修改哪里吗  这个冻结训练我之前没用过 一共100个epoch 在第50个epoch的时候报错中断了 我看那个train里面有个设置的冻结的50    是不是就把model_path换成最后第50轮保存的那个权重再次进行训练就行 还是要怎么操作    感谢感谢感谢感谢感谢感谢非常的感谢

**Comment by leoda1** [comment]
对的 model_path加载这个权重 Init_Epoch改成50 就会从第50个epoch开始

**Comment by leoda1** [comment]
```python
Cuda = True
    distributed = False
    sync_bn = False
    fp16 = True
    num_classes = 11
    phi = "b2"
    pretrained = False
    model_path = "logs/last_epoch_weights.pth"
    input_shape = [720, 720]
    Init_Epoch = 50
    Freeze_Epoch = 50
    Freeze_batch_size = 16
    UnFreeze_Epoch = 100
    Unfreeze_batch_size = 8
    Freeze_Train = True
    Init_lr = 1e-4
    Min_lr = Init_lr * 0.01
    optimizer_type = "adamw"
    momentum = 0.9
    weight_decay = 1e-2
    lr_decay_type = 'cos'
    save_period = 20
    save_dir = 'logs'
    eval_flag = True
    eval_period = 10
    dataset = 'dataset'
    dice_loss = False
    focal_loss = False
```
可以改的挺多的，input_shape可以调小训练的快很多，optimizer_type可以改adam和adamw，phi可以改成b0, b1, b2（这里三个不同参数的backbone），学习率动量具体设置好的参数就不建议动了。

**Comment by JAYCHUS** [comment]
收到 收到  非非非非非非常感谢

**Comment by JAYCHUS** [comment]
请问您有训练好的aeroscapes的权重吗 可以发我一份吗      非非非非非非常感谢

**Comment by JAYCHUS** [comment]
还有就是 我看这您的train里面写的model_path  直接就是   model_path = "logs/last_epoch_weights.pth" 这个了  这个是不是刚开始训练的时候要加载 UTFormer里面的权重呀 这个可以发一下么 就是class  mit_b0    mit_b1  mit_b2 下面的加载的这个backbone的权重          if pretrained:
            print("Load backbone weights")
            self.load_state_dict(torch.load("model_weights/segformer_b0_backbone_weights.pth"), strict=False)

**Comment by leoda1** [comment]
pretrained我设置的False，那部分代码还是segformer的，UTFormer是从0开始训练的。

**Comment by JAYCHUS** [comment]
感谢回复  好的好的 我在跑一下  感谢感谢

**Comment by JAYCHUS** [comment]
Warning: grad and param do not obey the gradient layout contract. This is not an error, but may impair performance. 请问您遇到过这个问题吗 老天了 第二轮报这个错误 训练了30多个epoch 损失只有0.006了 很吓人 然而iou很低

**Comment by leoda1** [comment]
我没遇到 你这是loss收敛了但是miou很低吗？

**Comment by leoda1** [comment]
fp16那个混合精度训练你关了试试，另外（你可以把你数据集copy一份，留个里面1/4的数据测试） aeroscapes这个数据集一直很奇怪。我当时跑了十几次，每次结果都大相径庭。

**Comment by JAYCHUS** [comment]
昨天晚上走的时候跑的  早上来一看 miou是0  我刚试了把测试数据的那个txt减少了一半  好像还是不行 第二轮的时候 报错 Warning: grad and param do not obey the gradient layout contract. This is not an error, but may impair performance然后10个epoch就已经收敛了 .跟昨天一样  我去搜一下这个warning 怎么解决 另外可以发我一份你训练好的权重吗 我想看一下这个效果

**Comment by JAYCHUS** [comment]
感谢感谢

**Comment by JAYCHUS** [comment]
请问可以发我一份训练好的权重吗  改了一下午还是不行

**Comment by leoda1** [comment]
你有其他联系方式吗

**Comment by JAYCHUS** [comment]
我的qq 2710840551 哇真的太感谢了 谢谢谢谢谢谢您

**Comment by leoda1** [comment]
有微信吗

**Comment by JAYCHUS** [comment]
也可以加我的微信15838979736 或者您留您的联系方式我加您也行
